erpnext/accounts/doctype/account/account.js

```diff
@@ -22,9 +22,6 @@ frappe.ui.form.on("Account", {
 		// hide fields if group
 		frm.toggle_display(["tax_rate"], cint(frm.doc.is_group) == 0);
 
-		// disable fields
-		frm.toggle_enable(["is_group", "company"], false);
```

Comment:
These fields cannot be editable after save.

frm.toggle_enable(["is_group", "company", "account_number"], frm.is_new());